modules/402-ingress-nginx/hooks/safe_daemonset_update.go

```diff
@@ -161,7 +163,7 @@ func safeControllerUpdate(input *go_hook.HookInput, dc dependency.Container) (er
 		for _, p := range proxys {
 			proxy := p.(IngressFilterResult)
 			if (controller.Name == proxy.Name) && (controller.Checksum == proxy.Checksum) {
-				if proxy.Status.NumberReady == proxy.Status.CurrentNumberScheduled {
+				if proxy.Status.NumberAvailable == proxy.Status.CurrentNumberScheduled {
```

Comment:
A comment should follow this change because it is not apparent which daemonset status fields should be used and why.

Reply:
Done